valkey.conf

```diff
@@ -1439,6 +1439,12 @@ lazyfree-lazy-user-flush yes
 #
 # prefetch-batch-max-size 16
 #
+# For use cases where command replies include Bulk strings (e.g. GET, MGET)
+# reply offload can be enabled to eliminate espensive memory access
+# and redundant data copy performed by main thread
+#
+# reply-offload yes
```

Comment:
Do we expect their to be cases where tuning this variable makes sense? Generally we want to avoid configuration in Valkey to make it simple to operate. Can we make real-time decisions about offloading?

Reply:
I'd prefer to avoid the config too. It's better to start with no config and, if it turns out we need it later, then we can add. The reverse is not possible because removing a config is a breaking change.

Reply:
Please see results of **Performance tests** in the PR description. Reply offload benefits performance if either data size is large (e.g. 64 Kb) or number of I/O threads is big enough for small data sizes (e.g. starting 9 `io-threads` config for 512 byte). 

As eliminating expensive memory access to `obj->ptr ` by main thread is major component of reply offload optimization , it is very challenging to provide dynamic solution.  Note: access to `obj->ptr ` is required to know object size. Besides this, assuming object size is available somehow, it will be relatively challenging to calibrate IsReplyOffloadBeneficial(data_size, io_threads_num) to make it generic for any OS/CPU architecture/etc.

It looks like we should provide config parameter in this case and customer need to test their workloads with and without reply offload and decide to activate it or not.

Please share your opinions.

Reply:
We have other places where we dynamically choose to take an action, like with lazy-free effort and freeing objects in background threads. For lazy-free, we try to guess how much work it will be to free, and if the effort is small we do it in the main thread anyways. It seems like we can have a similar heuristic here, if the object is large or we have enough I/O threads, we offload the items to be freed. That way end users don't have to tune it and it works well by default.